src/writers.ts

```diff
@@ -1,22 +1,48 @@
 import { evm } from '@snapshot-labs/checkpoint';
 import { Payment } from '../.checkpoint/models';
+import { getJSON } from './utils';
+import tokens from './payment_tokens.json';
+
+function getTokenSymbol(tokenAddress: string, chain: string) {
+  return tokens[chain][tokenAddress];
+}
 
 export function createEvmWriters(indexerName: string) {
   const handlePaymentReceived: evm.Writer = async ({ block, tx, event }) => {
     console.log('In handlePaymentReceived');
     if (!block || !event) return;
 
     const sender = event.args.sender;
-    const token = event.args.token;
-    const amount = event.args.amount;
+    const tokenAddress = event.args.token.toLowerCase();
+    const amountRaw = event.args.amount.toString();
+    const amountDecimal = (Number(amountRaw) / 1e6).toFixed(2); // Transforms 1990000 to 1.99. We used 1e6 because USDC and USDT have 6 decimals.
     const barcode = event.args.barcode;
 
-    const payment = new Payment(`${sender}/${tx.hash}`, indexerName);
+    const chain = indexerName; // The indexer name corresponds to the chain name
+    let tokenSymbol = getTokenSymbol(tokenAddress, chain);
+
+    if (!tokenSymbol) {
+      tokenSymbol = tokenAddress;
+    }
```

Comment:
I would not store the address as token symbol, this would require adding extra logic in the UI to detect if there is a real symbol that we can show to user or if it's a long address.
```suggestion
    let tokenSymbol = getTokenSymbol(tokenAddress, chain) || '';
```